### PR TITLE
docs: add documentation for influxdb scaler

### PR DESCRIPTION
**Pull request by yquansah**
Related to: https://github.com/kedacore/keda/issues/1239

### CODE REVIEW

File: content/docs/2.0/scalers/influxdb.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
++++
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:1]: Nit - The official name is `InfluxDB`, can we rename all `Influxdb` to this please?

File: content/docs/2.0/scalers/influxdb.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+### Trigger Specification
+
+This specification describes the `influxdb` trigger that scales based on the results of a flux query.
+
+```yaml
+triggers:
+  - type: influxdb
+    metadata:
+      authToken: some-auth-token
+      organizationName: influx-org
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:18]: Does this scaler only support InfluxDB Cloud or not? Because then we should rename the scaler imo

File: content/docs/2.0/scalers/influxdb.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      query: |
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+```
+
+**Parameter list:**
+
+- The `authToken`/`authTokenFromEnv` field(s) holds the authentication token needed for the influxdb client to communicate with an associated server. `authTokenFromEnv` should hold the name of the environment variable in the target deployment that contains the actual authorization token value.
+- `organizationName`/`organizationNameFromEnv` field(s) holds another value needed by the client. `organizationNameFromEnv` has the same semantics of pulling a value from the target deployment as `authTokenFromEnv`.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:32]: This is not very clear what this represents, can you elaborate on why I would need to specify this as a user? Is this the org name in InfluxDB Cloud?

File: content/docs/2.0/scalers/influxdb.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- `serverURL` holds the url value of the influxdb destination server.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:33]:
```suggestion
- `serverURL` holds the url value of the influxdb server.
```

File: content/docs/2.0/scalers/influxdb.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+- `thresholdValue` is provided by the user. This value can vary from use case to use case depending on the data of interest.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:34]: You should elaborate here that this is the value that will trigger the scaling

File: content/docs/2.0/scalers/influxdb.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+- `query` is the flux query that will yield the value for the scaler to compare the `thresholdValue` against.
+
+### Example
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:37]: Can you add a section on authentication parameters similar to https://github.com/kedacore/keda-docs/blob/master/content/docs/2.0/scalers/azure-service-bus.md#authentication-parameters please?

This typically describes the scenarios, in this case auth token, and if it supports TriggerAuthentication or not

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Thanks for the PR! Added some remarks to make it easier for users to understand what goes where

File: content/docs/2.0/scalers/influxdb.md (hunk 2 above)
Comment (yquansah) [reply content/docs/2.0/scalers/influxdb.md:18]: No it supports open source version as well

Comment (yquansah) [reply content/docs/2.0/scalers/influxdb.md:18]: https://docs.influxdata.com/influxdb/v2.0/organizations/

File: content/docs/2.0/scalers/influxdb.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      query: |
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+```
+
+**Parameter list:**
+
+- The `authToken`/`authTokenFromEnv` field(s) holds the authentication token needed for the influxdb client to communicate with an associated server. `authTokenFromEnv` should hold the name of the environment variable in the target deployment that contains the actual authorization token value.
+- `organizationName`/`organizationNameFromEnv` field(s) hold the organization name needed for the client to locate all [information](https://docs.influxdata.com/influxdb/v2.0/organizations/) contained in that organizaation such as buckets, tasks, etc. `organizationNameFromEnv` has the same semantics of pulling a value from the target deployment as `authTokenFromEnv`.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:32]:
```suggestion
- `organizationName`/`organizationNameFromEnv` field(s) hold the organization name needed for the client to locate all information contained in that [organization](https://docs.influxdata.com/influxdb/v2.0/organizations/) such as buckets, tasks, etc. `organizationNameFromEnv` has the same semantics of pulling a value from the target deployment as `authTokenFromEnv`.
```

File: content/docs/2.0/scalers/influxdb.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+      organizationNameFromEnv: INFLUXDB_ORG_NAME
+      serverURL: http://influxdb:8086
+      thresholdValue: '4'
+      query: |
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+```
+
+**Parameter list:**
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:29]: Nit again but I would seperately list the `FromEnv` entries from the non-env ones and just mention that it does the same but read from an env variable.

For example:
```markdown
- `organizationName` is the organization name needed for the client to locate all [information](https://docs.influxdata.com/influxdb/v2.0/organizations/) contained in that organization such as buckets, tasks, etc.
- `organizationNameFromEnv` defines the organization name, similar to `organizationName`, but reads it from an environment variable on the scale target.
```

File: content/docs/2.0/scalers/influxdb.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+- `thresholdValue` is provided by the user. This value can vary from use case to use case depending on the data of interest, and is needed to trigger the scaling in/out depending on what value comes back from the query.
+- `query` is the flux query that will yield the value for the scaler to compare the `thresholdValue` against.
+
+### Authentication Parameters
+
+You can authenticate by using an authorization token.
+
+**Authorization Token Authentication:**
+
+- `authToken` - Authorization token for influx db server.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:43]:
```suggestion
- `authToken` - Authorization token for InfluxDB server.
```

File: content/docs/2.0/scalers/influxdb.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+```
+
+**Parameter list:**
+
+- The `authToken`/`authTokenFromEnv` field(s) holds the authentication token needed for the influxdb client to communicate with an associated server. `authTokenFromEnv` should hold the name of the environment variable in the target deployment that contains the actual authorization token value.
+- `organizationName`/`organizationNameFromEnv` field(s) hold the organization name needed for the client to locate all [information](https://docs.influxdata.com/influxdb/v2.0/organizations/) contained in that organizaation such as buckets, tasks, etc. `organizationNameFromEnv` has the same semantics of pulling a value from the target deployment as `authTokenFromEnv`.
+- `serverURL` holds the url value of the influxdb server.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:33]:
```suggestion
- `serverURL` holds the url value of the InfluxDB server.
```

File: content/docs/2.0/scalers/influxdb.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      thresholdValue: '4'
+      query: |
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+```
+
+**Parameter list:**
+
+- The `authToken`/`authTokenFromEnv` field(s) holds the authentication token needed for the influxdb client to communicate with an associated server. `authTokenFromEnv` should hold the name of the environment variable in the target deployment that contains the actual authorization token value.
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/influxdb.md:31]:
```suggestion
- The `authToken`/`authTokenFromEnv` field(s) holds the authentication token needed for the InfluxDB client to communicate with an associated server. `authTokenFromEnv` should hold the name of the environment variable in the target deployment that contains the actual authorization token value.
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
A few more remarks but almost there, thanks!

File: content/docs/2.0/scalers/influxdb.md (hunk 2 above)
Comment (tomkerkhove) [reply content/docs/2.0/scalers/influxdb.md:18]: Makes sense, thanks!

File: content/docs/2.0/scalers/influxdb.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
++++
+title = "InfluxDB"
+layout = "scaler"
+availability = "v2.0+"
```
[the remark this answers is not in the document]
Comment (zroubalik) [reply content/docs/2.0/scalers/influxdb.md:4]: We should prepare 2.1 directories in the docs first and this PR should target that instead of 2.0

[the remark this answers is not in the document]
Comment (tomkerkhove) [reply content/docs/2.0/scalers/influxdb.md:4]: Good point 👉 https://github.com/kedacore/keda-docs/pull/316

**Review by zroubalik — CHANGES_REQUESTED** [review]
@yquansah could you please target this against 2.1? ie. `content/docs/2.1/scalers/`

**Comment by yquansah** [comment]
@tomkerkhove Done! Thank you for your detailed reviews, they are much appreicated.

**Review by tomkerkhove — APPROVED** [review]
LGTM, ready to be merged when the feature is merged @zroubalik

File: content/docs/2.1/scalers/influxdb.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+This specification describes the `InfluxDB` trigger that scales based on the results of a flux query.
+
+```yaml
+triggers:
+  - type: influxdb
+    metadata:
+      serverURL: http://influxdb:8086
+      organizationName: influx-org
+      organizationNameFromEnv: INFLUXDB_ORG_NAME
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/influxdb.md:19]: Can we comment this one out, or at least mention that you either have to choose between `organizationName` or `organizationNameFromEnv` and not both please?

File: content/docs/2.1/scalers/influxdb.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+      thresholdValue: '4'
+      query: |
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+      authToken: some-auth-token
```
Comment (tomkerkhove) [inline content/docs/2.1/scalers/influxdb.md:25]: Can we comment this one out, or at least mention that you either have to choose between `authToken` or `authTokenFromEnv` and not both please?

File: content/docs/2.1/scalers/influxdb.md (hunk 13 above)
Comment (yquansah) [reply content/docs/2.1/scalers/influxdb.md]: @tomkerkhove Yeah no problem. I was following the Redis List example here: https://keda.sh/docs/2.0/scalers/redis-lists/

Which is why I added both, but I will comment the `env` ones out.

**Comment by yquansah** [comment]
@tomkerkhove Thanks a ton for the review 🙏

File: content/docs/2.1/scalers/influxdb.md (hunk 13 above)
Comment (tomkerkhove) [reply content/docs/2.1/scalers/influxdb.md]: Thanks! Will update those asap then to align as well.

**Comment by tomkerkhove** [comment]
Thank you for your contribution!

File: content/docs/2.1/scalers/influxdb.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+- `authTokenFromEnv` defines the authorization token, similar to `authToken`, but reads it from an environment variable on the scale target.
+- `organizationName` is the organization name needed for the client to locate all information contained in that [organization](https://docs.influxdata.com/influxdb/v2.0/organizations/) such as buckets, tasks, etc.
+- `organizationNameFromEnv` defines the organization name, similar to `organizationName`, but reads it from an environment variable on the scale target.
+- `serverURL` holds the url value of the InfluxDB server.
+- `thresholdValue` is provided by the user. This value can vary from use case to use case depending on the data of interest, and is needed to trigger the scaling in/out depending on what value comes back from the query.
+- `query` is the flux query that will yield the value for the scaler to compare the `thresholdValue` against.
+- `metricName` is an optional parameter that a user can give to provide some uniqueness of `ScaledObject` names. It's optional and will default in source code if not provided as a parameter.
```
Comment (zroubalik) [inline content/docs/2.1/scalers/influxdb.md:39]: Minor nit here, could you please use the similar description (and examples) for `metricName` that's being used in this PR: https://github.com/kedacore/keda-docs/pull/313

**Review by zroubalik — CHANGES_REQUESTED** [review]

File: content/docs/2.1/scalers/influxdb.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+- `authTokenFromEnv` defines the authorization token, similar to `authToken`, but reads it from an environment variable on the scale target.
+- `organizationName` is the organization name needed for the client to locate all information contained in that [organization](https://docs.influxdata.com/influxdb/v2.0/organizations/) such as buckets, tasks, etc.
+- `organizationNameFromEnv` defines the organization name, similar to `organizationName`, but reads it from an environment variable on the scale target.
+- `serverURL` holds the url value of the InfluxDB server.
+- `thresholdValue` is provided by the user. This value can vary from use case to use case depending on the data of interest, and is needed to trigger the scaling in/out depending on what value comes back from the query.
+- `query` is the flux query that will yield the value for the scaler to compare the `thresholdValue` against.
+- `metricName` is an optional parameter that a user can give to provide uniqueness of `ScaledObject` names. If not set KEDA will generate a unique name based on time. It is required that all metric names are unique, if using more than one trigger.
```
Comment (zroubalik) [inline content/docs/2.1/scalers/influxdb.md:39]: when reading this I have found out a problem, that I didn't spot in the original PR: 
https://github.com/kedacore/keda/blob/af2b36f593c231ed25c48a0ba432f6128f77cbf5/pkg/scalers/influxdb_scaler.go#L77

this is not ideal, the metric name should be immutable, ie. timestamp is not the right way. Could you please revert it and base in on organization name and/or serverURL (the same way it is done for postgresql scaler?) Thanks!

**Review by zroubalik — CHANGES_REQUESTED** [review]

File: content/docs/2.1/scalers/influxdb.md (hunk 16 above)
Comment (zroubalik) [reply content/docs/2.1/scalers/influxdb.md:39]: I am sorry I didn't find this when doing the review on the original PR

Comment (yquansah) [reply content/docs/2.1/scalers/influxdb.md:39]: @zroubalik No worries. So the serverURL and organization name is not enough to make it unique. I can take a look into how Postgres does it.

Comment (zroubalik) [reply content/docs/2.1/scalers/influxdb.md:39]: @ycabrer yeah, but we should do the best effort for that and we can expect that majority of ScaledObjects will use just one scaler. We are implementing a solution for check metric name uniqueness on general level: https://github.com/kedacore/keda/pull/1390

Comment (zroubalik) [reply content/docs/2.1/scalers/influxdb.md:39]: @yquansah any update on the metric name please?

Comment (yquansah) [reply content/docs/2.1/scalers/influxdb.md:39]: @zroubalik Hello I've been slammed with work over the past week i apologize! I will get a PR in by today or tomorrow 👍

Comment (zroubalik) [reply content/docs/2.1/scalers/influxdb.md:39]: @yquansah no worries and no rush, just checking the status :)

Comment (yquansah) [reply content/docs/2.1/scalers/influxdb.md:39]: @zroubalik https://github.com/kedacore/keda/pull/1429 Please let me know feedback on this then I can update this PR as well

Comment (zroubalik) [reply content/docs/2.1/scalers/influxdb.md:39]: @yquansah thanks, I merged the PR, could you please rephrase this section a bit?

File: content/docs/2.1/scalers/influxdb.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+      organizationName: influx-org
+      organizationNameFromEnv: INFLUXDB_ORG_NAME # Optional: You can use this instead of `organizationName` parameter. See details in "Parameter List" section
+      thresholdValue: '4'
+      query: |
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+      authToken: some-auth-token
+      authTokenFromEnv: INFLUXDB_AUTH_TOKEN # Optional: You can use this instead of `authToken` parameter. See details in "Parameter List" section
+      metricName: influx-metric # Optional: This value will default to a masked version of the url if not set by the user (generated value would be `influxdb-influx-metric-influx-org`)
```
Comment (zroubalik) [inline content/docs/2.1/scalers/influxdb.md:27]:
```suggestion
      metricName: influx-metric # Optional: This value will default to a masked version of the url if not set by the user (generated value would be `influxdb-https---xxx-influx_org`)
```

**Review by zroubalik — CHANGES_REQUESTED** [review]
Small tweaks on `metricName` as per: https://github.com/kedacore/keda/pull/1444

File: content/docs/2.1/scalers/influxdb.md (hunk 17 above)
Comment (yquansah) [reply content/docs/2.1/scalers/influxdb.md:27]: No this is actually correct, because the `metricName` is provided. The value you suggest is a default.

Comment (yquansah) [reply content/docs/2.1/scalers/influxdb.md]: I will add documentation for when the value is defaulting though

File: content/docs/2.1/scalers/influxdb.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+      organizationName: influx-org
+      organizationNameFromEnv: INFLUXDB_ORG_NAME # Optional: You can use this instead of `organizationName` parameter. See details in "Parameter List" section
+      thresholdValue: '4'
+      query: |
+        from(bucket: "bucket-of-interest")
+        |> range(start: -12h)
+        |> filter(fn: (r) => r._measurement == "stat")
+      authToken: some-auth-token
+      authTokenFromEnv: INFLUXDB_AUTH_TOKEN # Optional: You can use this instead of `authToken` parameter. See details in "Parameter List" section
+      metricName: influx-metric # Optional: This value will default to a masked version of the url if not set by the user (generated value would be `influxdb-influx-metric-influx-org`). If this value isn't set the generated value would be (`influxdb-https---xxx-influx_org`)
```
Comment (zroubalik) [inline content/docs/2.1/scalers/influxdb.md:27]:
```suggestion
      metricName: influx-metric # Optional: This value will default to a masked version of the url and organization name if not set by the user (metrics name value would be then `influxdb-https---xxx-influx_org`).
```

I'd rephrase this a little bit, the previous version was a little bit confusing. imho.

**Review by zroubalik — APPROVED** [review]
LGTM, thanks a lot!
